### DISCUSSION TITLE
After switching to latest flask, app timeouts on some requests

### DISCUSSION THREAD

**Discussion by pipozzz** [pallets/flask#5132] [Q&A] [answered]
When I wanted to upgrade from 2.2.x to 2.3.x I encountered problem with timeouts on some requests, without any error log about it.

Any ideas please ? What can cause it?

- Python version: 3.10
- Flask version: 2.3.2
- gunicorn: 20.1.0
- gevent: 22.10.2
- nginx: 1.18.0

**Comment by davidism** [comment]
It is impossible to say with the lack of information. Be sure to provide a minimal reproducible example.

**Reply by pipozzz** [reply]
it's very hard to debug, because I see no errors just see that calls timeout, like broken event loop or so. In the frontend I see 499 error codes I think it's because of client connect timeout exceeded. One request is fine another timeouts. Everything started when I switched from 2.2.x to 2.3.x. 

here is diff in requirements.txt
```
diff 2.2.3.txt latest.txt         
30c30
< Flask==2.2.5
---
> Flask==2.3.2
108c108
< Werkzeug==2.2.3
---
> Werkzeug==2.3.4
```

what else would you need I can provide you.

Thanks in advance.

**Reply by pipozzz** [reply]
Could you please give me some suggestion how to debug this or so? I'd appreciate any idea.

**Reply by cplpearce** [reply]
@pipozzz We just recently upgraded from `Flask 2.2.2` to `latest` and I'm experiencing the same issue.

We use the following packages:
- `Flask-Bcrypt`
- `Flask-Cors`
- `pylint-flask`
- `flask-session2`
- `flask-compress`

I rolled Flask back to `2.2.2` and the issue has disappeared. I haven't had the time to debug it so I'm sorry I can't offer anything more!

**Reply by pipozzz** [reply, marked as answer]
I found my problem , It's combination of gunicorn + gevent + latest flask this is not working, I switched to uwsgi + gevent + latest flask and now it works
